R/http.R

```diff
@@ -354,6 +326,58 @@ queryString <- function(elements) {
   return(result)
 }
 
+buildUrl <- function(apiPath, path, query = NULL) {
+  # prepend the service path
+  url <- paste(apiPath, path, sep = "")
+
+  # append the query
+  if (!is.null(query)) {
+    # URL encode query args
+    query <- utils::URLencode(query)
+    url <- paste(url, "?", query, sep = "")
+  }
+
+  url
+}
+
+requestCertificate <- function(protocol, certificate = NULL) {
+  if (identical(protocol, "https")) {
+    createCertificateFile(certificate)
+  } else {
+    NULL
+  }
+}
+
+# Auth --------------------------------------------------------------------
+
+authHeaders1 <- function(authInfo, method, url, file) {
```

Comment:
Auth in `httpRequest` and `httpRequestWithBody` appears to have slightly drifted apart, but I think the two functions are equivalent, and since `authHeaders1()` is slightly simpler, I'd suggest standardising on it.

Reply:
Here is a path with certificate-only `authInfo`:
https://github.com/rstudio/rsconnect/blob/e9ec50cfc2c5c797afddf8f7ebe10cedc8b041ca/R/ide.R#L50-L53

Added in https://github.com/rstudio/rsconnect/commit/96ba75b944b0e35a88dc7b8a940233739b96eb4f

It looks like this is the only case where `authInfo` is not an account, but something else, and might be the only instance where it contains a certificate.

In other cases, we use `clientForAccount` to obtain a client, which augments the `account` object with a certificate.

It looks as if the other auth checks (those in `authHeaders1`) won't fire when the other fields (secret, private key, api key) are missing from `authInfo`. I agree; it looks like we can ditch `authHeaders2`.

Could we add tests to help document the fact that `authInfo` is usually more than an account (because `certificate` is added), but sometimes only has `certificate`?